ios/Classes/codec/AblyFlutterWriter.m

```diff
@@ -3,59 +3,69 @@
 #import "ARTTypes.h"
 #import "AblyFlutterMessage.h"
 #import "AblyFlutterReader.h"
+#import "AblyPlatformConstants.h"
 
 
 @implementation AblyFlutterWriter
 
 - (void)writeValue:(id)value {
     if([value isKindOfClass:[ARTErrorInfo class]]){
-        [self writeErrorInfo: value];
+        [self writeByte:errorInfoCodecType];
+        [self writeValue: [self encodeErrorInfo: value]];
```

Comment:
General observation but there is some inconsistency across your commits in respect of spaces after colons in message sending statements. I would prefer no spaces. So for this line, simply:

```suggestion
        [self writeValue:[self encodeErrorInfo: value]];
```